kroxylicious-filters/kroxylicious-encryption/doc/design.adoc

```diff
@@ -0,0 +1,267 @@
+= Encryption Filter Design
+Tom Bentley <tbentley@redhat.com>
+:toc:
+:icons: font
+:source-highlighter: pygments
+
+Design documentation on the encryption filter.
+
+== Scope
+
+The encryption filter is designed to provide confidentiality and integrity guarantees for data stored in Apache Kafka(TM), under the threat models described later in this document.
+
+Some assumptions are necessary in order to provide these guarantees, and these guarantees are not absolute. In other words, depending on how the proxy and filter are configured and deployed, it's possible that the actual security provided is less than what is described in this document. We will try to call-out in this document where we're making assumptions that affect the quality of security provided.
+
+== Threat models
+
+The aim is to provide additional security for data at-rest, from a threat actor who has operating system level access to the machines that comprise the Kafka cluster (i.e. brokers and controllers). We assume this actor can:
+
+* Read files on disk
+* Write files on disk
+* Obtain a core dump of broker and controller process memory at any time
+* Alter process memory at any time
+* Observe TCP packets
+* Craft TCP packets
+* Kill and restart processes
+
+We further assume that the threat actor
+
+* has network access to the client side of the Kafka proxy, and/or
+* has non-subverted access to existing Kafka client applications
+// What does the above mean, exactly?
+
+We assume the thread actor cannot interfere with proxy processes:
+
+* Reconfigure it (to remove)
+* Obtain a core dump of proxy process memory at any time
+
+== Key management
+
+=== Assumptions
+
+We assume the existence of a key management system (KMS) which has not been subverted by the threat actor. Specifically:
+
+. all communication between the proxy and KMS is protected TLS
+. that the KMS never leaks a key
```

Comment:
For consistency, use KEK rather than key.  Also define both KEK and DEK.

Reply:
Added a definitions table at the start of `Key management` with brief descriptions of the key types.